### PR TITLE
IMPRO-1713: Restore ignore_ecc_bounds option to PDF CLIs

### PR DESCRIPTION
**Pull request by bayliffe**
The ignore_ecc_bounds option is made available via the generate_realizations and generate_percentiles CLIs. This is primarily for use with global data.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new (restored old) tests for the new feature(s)

### CODE REVIEW

File: improver_tests/acceptance/test_generate_realizations.py [hunk 1]
```diff
@@ -113,6 +113,30 @@ def test_realizations(tmp_path):
...
 
 
+@pytest.mark.slow
+def test_ecc_bounds_warning(tmp_path):
+    """
+    Test use of ECC to convert one set of percentiles to another set of
+    percentiles, and then rebadge the percentiles to be ensemble realizations.
+    Data in this input exceeds the ECC bounds and so tests ecc_bounds_warning
+    functionality.
+    """
```
Comment (cgsandford) [inline improver_tests/acceptance/test_generate_realizations.py:123]: Is there a way to check the warning is raised?  Eg I think there is a `with pytest.warns` or something analagous available.

Comment (bayliffe) [reply improver_tests/acceptance/test_generate_realizations.py:123]: Yep. Done.

**Comment by bayliffe** [comment]
@cgsandford I have restored the functionality in both CLIs through simple fear that this issue will come up again as we work with more global model data. The whole reason we originally implemented all of this functionality in the underlying ECC code is that we have global model issues (e.g. erroneous supersonic wind speeds!). Whilst I don't have an immediate use case, I'd prefer to keep the functionality now that I have remembered why we need it.

I agree completely on the error message. Printing the whole array makes it rather useless. I've taken your suggestion and also rewritten the text as I find it rather hard to understand. Let me know if you think it is no clearer.

File: improver_tests/acceptance/test_generate_realizations.py (hunk 1 above)
Comment (cgsandford) [reply improver_tests/acceptance/test_generate_realizations.py:123]: Sorry I should have also commented on the other test file - I think that one merits a `pytest.warns` as well.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_generate_percentiles.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5518420&bid=18081502)

File: improver_tests/acceptance/test_generate_realizations.py (hunk 1 above)
Comment (gavinevans) [reply improver_tests/acceptance/test_generate_realizations.py:123]: I've added a check that a warning is raised using `pytest.warns` in `test_generate_percentiles.py`.

**Pull request merged**